core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -135,6 +138,26 @@ private[spark] class ExecutorAllocationManager(
   private val executor = Executors.newSingleThreadScheduledExecutor(
     Utils.namedThreadFactory("spark-dynamic-executor-allocation"))
 
+  // Metric source for ExecutorAllocationManager to expose the its internal executor allocation
+  // status to MetricsSystem.
+  private[spark] val executorAllocationManagerSource = new Source {
```

Comment:
Would you mind putting this as an inner class at the bottom (like `ExecutorAllocationManagerListener`) instead?